### DISCUSSION TITLE
Belchertown v1.3 and weewx 5 - it works!

### DISCUSSION THREAD

**Discussion by James-76** [poblabs/weewx-belchertown#903] [General]
Hello

At some point weewx 5 will be released, so I decided to take the leap and see what happens if you swapped over to using the latest beta (in this case 5.0.0b10) alongside the Belchertown extension

In my case I went down the pip route, and all of my issues were around making sure that the right python packages were installed in the virtual environment and dealing with file permissions as this isn't running as root.  Also dealing with the fact that weewx 5 needs 3.7 or later and my system was using python 3.6
 
This is what I ended up though I'm running some other customizations / user python extension scripts which are not installed as true extensions (metargenerator.py / fuzzy-archer)

```
(weewx-venv) [rocky@pi ~]$ weectl extension list
Extension Name    Version   Description
Belchertown       1.3       A clean modern skin with real time streaming updates and interactive charts. Modeled after BelchertownWeather.com
mqtt              0.24      Upload weather data to MQTT server.
```

```
(weewx-venv) [rocky@pi ~]$ pip list
Package            Version
------------------ -----------
certifi            2023.7.22
charset-normalizer 3.2.0
Cheetah3           3.2.6.post1
configobj          5.0.8
CT3                3.3.2
ephem              4.1.4
idna               3.4
paho-mqtt          1.6.1
Pillow             9.4.0
pip                23.2.1
PyMySQL            1.1.0
pyserial           3.5
pyusb              1.2.1
requests           2.31.0
setuptools         50.3.2
six                1.16.0
urllib3            2.0.4
weewx              5.0.0b10
wheel              0.41.1
```

But in summary it does all work without any changes to @poblabs code.

weewx 5 docs can be found [here](https://weewx.com/docs/5.0/)

James

**Comment by fvirgola80** [comment]
Hi James,
Could you please give a full description of the steps you took to upgrade to weewx 5 from 4.10.2 as a baseline with setup.py. I have the Raspberry model 3b+.
Could you give me a hand?
Thank you

**Reply by James-76** [reply]
Unfortunately since then I've done several changes, including moving away from pip and using the rpm install so I don't have any clear notes of what I did apart from just working through the online docs.

There is an online guide https://weewx.com/docs/5.0/upgrade/, however most of the issues I experienced were due to the level of OS and making sure you are at least running python 3.6 as default.

[rocky@pi ~]$ python -V
Python 3.9.18
